### ISSUE TITLE
Table size COUNT(*) for VIEWs is larger than that for TABLE created using the same VIEW from partitioned parquet dataset on s3

### ISSUE THREAD

**Issue by rajivpatki** [duckdb/duckdb#11280]
### What happens?

```

CREATE TEMPORARY VIEW _need_not_be_temp_view AS
(
    SELECT 
    *
    FROM read_parquet('s3://mybucket/dataset_name/date=*/domain=*/*.parquet')
    INNER JOIN my_existing_table_doesnt_matter USING (path, domain)
);

CREATE OR REPLACE TABLE table_from_need_not_be_temp_view AS
    SELECT * FROM _need_not_be_temp_view;

SELECT
path,
count(*)
FROM _need_not_be_temp_view
GROUP BY all;

```

Output:
```
index	path	count_star()
0	p1	880
1	p2	390
2	p3	390

	. . .

33	p34	6660
34	p35	700
```

But when I do:
```
SELECT path, COUNT(*) from table_from_need_not_be_temp_view group by all;
```

I get:
```
index	path	count_star()
0	p1	5
1	p2	21
2	p3	49
. . .
8	p9	203
9	p10	22
```
ie: Way fewer rows and the counts are lower too.

We tried doing the above with `TEMP TABLE` and `VIEW` too but are facing the same issue. Note that `path` is of type `list[str]` (python)

### To Reproduce

Use the above queries with a dataset that is partitioned in the format `s3://mybucket/dataset_name/date=*/domain=*/*.parquet`

### OS:

x86

### DuckDB Version:

0.10.1

### DuckDB Client:

Python

### Full Name:

Rajiv Patki

### Affiliation:

Jodaro

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have not tested with any build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [x] Yes, I have

**Comment by Mytherin** [comment]
Thanks for the report! I have managed to reproduce this locally - it seems there is an optimizer misbehaving. Toggling `pragma disable_optimizer;` fixes the problem for me. I will have a look at solving this.

**Comment by Mytherin** [comment]
This should be fixed in https://github.com/duckdb/duckdb/pull/11340 - there was an issue with joining/grouping by arrays where the hash would be computed incorrectly

**Comment by rajivpatki** [comment]
> This should be fixed in #11340 - there was an issue with joining/grouping by arrays where the hash would be computed incorrectly

Thank you so much! Your team's work is phenomenal
